packages/flutter_tools/lib/src/web/bootstrap.dart

```diff
@@ -13,13 +13,74 @@ import 'package:package_config/package_config.dart';
 /// This file is served when the browser requests "main.dart.js" in debug mode,
 /// and is responsible for bootstrapping the RequireJS modules and attaching
 /// the hot reload hooks.
+///
+/// If `generateLoadingIndicator` is true, embeds a loading indicator onto the
+/// web page that's visible while the Flutter app is loading.
 String generateBootstrapScript({
   required String requireUrl,
   required String mapperUrl,
+  required bool generateLoadingIndicator,
 }) {
   return '''
 "use strict";
 
+${generateLoadingIndicator ? _generateLoadingIndicator() : ''}
```

Comment:
does it matter that the loading indicator code will execute first now?

Reply:
otherwise lgtm

Reply:
This PR is not changing it. It was executing first before this PR.

Reply:
you're right. i suck at reading diffs, LGTM